### PR TITLE
Add jupyter server shutdown timeout setting

### PR DESCRIPTION
**Pull request by lucyzhang929**
Follow-up to this PR https://github.com/microsoft/azuredatastudio/pull/15899. I'm setting the default timeout back to 5 minutes and will add the setting in the smoke test repo and configure to 10 minutes.

### CODE REVIEW

**Review by barbaravaldez — APPROVED** [review]

File: extensions/notebook/src/jupyter/jupyterNotebookProvider.ts [hunk 1]
```diff
@@ -64,13 +64,15 @@ export class JupyterNotebookProvider implements nb.NotebookProvider {
 				manager.sessionManager.shutdown(session.id);
 			}
 			if (sessionManager.listRunning().length === 0) {
-				const TenMinutesInMs = 10 * 60 * 1000;
+				let notebookConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(constants.notebookConfigKey);
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/jupyterNotebookProvider.ts:67]: I'm thinking we should have 0 be "never shuts down" as well (no timeout)

File: extensions/notebook/package.json [hunk 2]
```diff
@@ -43,6 +43,11 @@
           "default": false,
           "description": "%notebook.dontPromptPythonUpdate.description%"
         },
+        "notebook.jupyterServerShutdownTimeout": {
+          "type": "number",
```
Comment (Charles-Gagnon) [inline extensions/notebook/package.json:47]: I think you should be able to set a min here, so we should not allow negative numbers (see comment below about 0)

File: extensions/notebook/src/jupyter/jupyterNotebookProvider.ts (hunk 1 above)
Comment (lucyzhang929) [reply extensions/notebook/src/jupyter/jupyterNotebookProvider.ts:67]: I think that makes sense. The servers will get shutdown once ADS is closed so there shouldn't be any issues with not having a timeout.

File: extensions/notebook/package.json (hunk 2 above)
Comment (lucyzhang929) [reply extensions/notebook/package.json:47]: I couldn't find any examples of setting the min in other extension settings and tried to use "min" here but that didn't work. With the new logic, if any negative numbers are entered by accident, there will be no timeout.

Comment (barbaravaldez) [reply extensions/notebook/package.json:47]: I think there is a **minimum** setting that you can use.

Comment (Charles-Gagnon) [reply extensions/notebook/package.json:47]: Try minimum

Comment (lucyzhang929) [reply extensions/notebook/package.json:47]: Ah yes that works! Thanks :)

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**